flowise-embed/src/components/Bot.tsx

```diff
@@ -1021,7 +1021,13 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
       if (warningMessage) {
         message += `\n**${warningMessage}**\n`;
       }
-      for (const ncm of ncmArray) {
+
+      const criticalAnalysisFourthStep = prefix === CriticalAnalysisPrefixes.criticalAnalysisFourthStep;
+      const criticalAnalysisFifthStep = prefix === CriticalAnalysisPrefixes.criticalAnalysisFifthStep;
+      const limitedCriticalAnalysis = criticalAnalysisFourthStep || criticalAnalysisFifthStep;
+      const ncmList = limitedCriticalAnalysis ? [ncmArray[0]] : ncmArray;
```

Comment:
```
const singleExecutionSteps = [CriticalAnalysisPrefixes.criticalAnalysisFourthStep, CriticalAnalysisPrefixes.criticalAnalysisFifthStep];
const isStepOfSingleExecution = singleExecutionSteps.includes(prefix);
const ncmList = isStepOfSingleExecution ? [ncmArray[0]] : ncmArray;

Reply:
Outra opção seria colocar todo o conteúdo de dentro do `for` em uma função e chamá-la de acordo com a necessidade - então no caso das etapas de execução única, apenas chamar uma vez e, nos demais, aí sim iniciar o `for` e chamá-la diversas vezes